Make pyawaitable.h support precompiled headers. (#46)
ZeroIntensity/pyawaitable: docs/installation.md

| `pyawaitable_abi`            | `PyAwaitable_ABI`             |
| `PyAwaitableType`            | `PyAwaitable_Type`            |

## Using Precompiled Headers with PyAwaitable

> Using precompiled headers in a large C extension project is supported, just replace `PYAWAITABLE_THIS_FILE_INIT` with `PYAWAITABLE_USE_PCH` **BEFORE** including your precompiled header file in every source file.

Reviewer: ```suggestion
Using precompiled headers in a large C extension project is supported, just replace `PYAWAITABLE_THIS_FILE_INIT` with `PYAWAITABLE_USE_PCH` before including your precompiled header file in every source file.
```

> Using precompiled headers in a large C extension project is supported, just replace `PYAWAITABLE_THIS_FILE_INIT` with `PYAWAITABLE_USE_PCH` ***BEFORE*** including your precompiled header file in every source file.

Author: ```suggestion
Using precompiled headers in a large C extension project is supported, just replace `PYAWAITABLE_THIS_FILE_INIT` with `PYAWAITABLE_USE_PCH` **BEFORE** including your precompiled header file in every source file.
```

> After that inside of the precompiled header file right after including `pyawaitable.h` add this line:

Reviewer: ```suggestion
After that, inside of the precompiled header file--right after including `pyawaitable.h`--add this line:
```

> After that inside of the precompiled header file right after it includes `pyawaitable.h` add this line:

Author: ```suggestion
After that inside of the precompiled header file right after including `pyawaitable.h` add this line:
```

> After that in each non-module init C file this should be done **AFTER** including your precompiled header file:

Author: ```suggestion
After that inside of the precompiled header file right after it includes `pyawaitable.h` add this line:
```

> After that in each non-module init C file this should be done ***AFTER*** including your precompiled header file:

Author: ```suggestion
After that in each non-module init C file this should be done **AFTER** including your precompiled header file:
```

```c
DECLARE_PYAWAITABLE_ABI;
```

In the source file with the module initialization code:

```c
PYAWAITABLE_INIT_DEF;
```

> And then finally after the include of `pch.h` inside of `pch.c` (`pch` is placeholder name for the precompiled header):

Reviewer: ```suggestion
And then, finally after the include of `pch.h`--inside of `pch.c` (`pch` is placeholder name for the precompiled header):
```

```c
DECLARE_PYAWAITABLE_ABI = NULL;
```

> After that all logic from the example code from `test.c` still applies.

Author: ```suggestion
After that, all logic from the example code in `test.c` still applies.
```
